src/components/3d/SimpleModelView.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import {
+  Animated,
+  PanResponder,
+  StyleSheet,
+  View,
+  ViewStyle,
+} from 'react-native';
+import { WebView } from 'react-native-webview';
+import { ImgixImage } from '@rainbow-me/images';
+
+export type ModelViewerProps = {
+  readonly setLoading: (loading: boolean) => void;
+  readonly loading: boolean;
+  readonly style?: ViewStyle;
+  readonly uri?: string;
+  readonly alt?: string;
+  readonly fallbackUri?: string;
+};
+
+const styles = StyleSheet.create({
+  bottomRight: {
+    alignItems: 'flex-end',
+    justifyContent: 'flex-end',
+    padding: 10,
+  },
+  flex: { flex: 1 },
+});
+
+export default function ModelViewer({
+  loading,
+  setLoading,
+  style,
+  uri,
+  alt,
+  fallbackUri,
+}: ModelViewerProps): JSX.Element {
+  const opacity = React.useMemo(() => new Animated.Value(1), []);
+  const [visibility, setVisibility] = React.useState<boolean>(false);
+  const [progress, setProgress] = React.useState<number>(0);
+  const originWhiteList = React.useMemo(() => ['*'], []);
+  const onMessage = React.useCallback(
+    ({ nativeEvent: { data } }) => {
+      const { type, payload } = JSON.parse(data);
+      if (type === 'progress') {
+        setProgress(payload as number);
+      } else if (type === 'model-visibility') {
+        setVisibility(payload as boolean);
+      }
+    },
+    [setProgress, setVisibility]
+  );
+  const fallbackSource = React.useMemo(
+    () => ({
+      uri: fallbackUri,
```

Comment:
@cawfree let's catch up about webview, react-native-video and react-native-sound before merging this

Reply:
@osdnk We're currently just finishing up some supporting work here, but this will happen in a separate PR.